File: api/booking-service/src/main/java/com/carrental/bookingservice/model/dto/VehicleDetailsDTO.java
package com.carrental.bookingservice.model.dto;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@Data
public class VehicleDetailsDTO {

    @NotEmpty
    @Size(max = 50)
    private String bodyType;

    @NotNull
    private Integer productionYear;

    @NotEmpty
    @Size(max = 50)
    private String fuelType;

    @NotNull
    private Integer power;

    @NotEmpty
    @Size(max = 50)
    private String gearbox;

    @NotNull
    private boolean frontWheelDrive;

    @NotNull
    private Integer doorsNumber;

    @NotNull
    private Integer seatsNumber;

    @NotEmpty
    @Size(max = 50)
    private String color;

    @NotNull
    private boolean metallic;

    @NotEmpty
    @Size(max = 70)
    private String imageName;

    @NotEmpty
    @Size(max = 255)
    private String description;
}

File: frontend-app/shared/model/VehicleDetailsDTO.ts
export class VehicleDetailsDTO {
    bodyType: string;
    productionYear: number;
    fuelType: string;
    power: number;
    gearbox: string;
    frontWheelDrive: boolean;
    doorsNumber: number;
    seatsNumber: number;
    color: string;
    metallic: boolean;
    imageName: string;
    description: string;

    constructor(
        bodyType: string,
        productionYear: number,
        fuelType: string,
        power: number,
        gearbox: string,
        frontWheelDrive: boolean,
        doorsNumber: number,
        seatsNumber: number,
        color: string,
        metallic: boolean,
        imageName: string,
        description: string
    ) {
        this.bodyType = bodyType;
        this.productionYear = productionYear;
        this.fuelType = fuelType;
        this.power = power;
        this.gearbox = gearbox;
        this.frontWheelDrive = frontWheelDrive;
        this.doorsNumber = doorsNumber;
        this.seatsNumber = seatsNumber;
        this.color = color;
        this.metallic = metallic;
        this.imageName = imageName;
        this.description = description;
    }
}
